### PR TITLE
Add DirichletMinkowski GH BC

### PR DESCRIPTION
**Pull request by geoffrey4444**
## Proposed changes

Add a GH Dirichlet boundary condition that sets the spacetime metric, phi, and pi to Minkowski. This allows you to impose flat space on the outer boundary without supplying an analytic solution, which is necessary for testing Dirichlet boundary conditions in binary-black-hole simulations, which do not have an analytic solution.

Ultimately, we will want to use bjorhus, constraint-preserving boundary conditions in these simulations, but until simulations using the Bjorhus BCs are working, having a flat-space Dirichlet condition as an alternative is helpful for comparing with the Bjorhus conditions.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/DirichletMinkowski.hpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+/// \cond
+namespace domain::Tags {
+template <size_t Dim, typename Frame>
+struct Coordinates;
+}  // namespace domain::Tags
+/// \endcond
+
+namespace GeneralizedHarmonic::BoundaryConditions {
+/*!
+ * \brief Sets Dirichlet boundary conditions using the analytic solution or
```
Comment (kidder) [inline src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/DirichletMinkowski.hpp]: Sets Dirichlet boundary conditions corresponding to a Minkowski spacetime

File: src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/DirichletMinkowski.hpp [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+          tnsr::I<DataVector, Dim, Frame::Inertial>>& /*face_mesh_velocity*/,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& /*normal_covector*/,
+      const tnsr::I<DataVector, Dim, Frame::Inertial>& /*normal_vector*/,
+      const tnsr::I<DataVector, Dim, Frame::Inertial>& coords,
+      const Scalar<DataVector>& interior_gamma1,
+      const Scalar<DataVector>& interior_gamma2,
+      const double time) const noexcept {
+    *gamma1 = interior_gamma1;
+    *gamma2 = interior_gamma2;
+    auto boundary_values = [&coords, &time]() noexcept {
```
Comment (kidder) [inline src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/DirichletMinkowski.hpp]: just set everything here to their analytical values without calling into the analytic solution, or using the other free functions.  Then you don't need to specify the coordinates or time in the tags above or take them as arguments

File: src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/DirichletMinkowski.hpp [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+          inv_spatial_metric,
+      const std::optional<
+          tnsr::I<DataVector, Dim, Frame::Inertial>>& /*face_mesh_velocity*/,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& /*normal_covector*/,
+      const tnsr::I<DataVector, Dim, Frame::Inertial>& /*normal_vector*/,
+      const tnsr::I<DataVector, Dim, Frame::Inertial>& coords,
+      const Scalar<DataVector>& interior_gamma1,
+      const Scalar<DataVector>& interior_gamma2,
+      const double time) const noexcept {
+    *gamma1 = interior_gamma1;
```
Comment (kidder) [inline src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/DirichletMinkowski.hpp]: should be able to move the definition of this function to the cpp file

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by geoffrey4444** [comment]
Rebased and pushed a fixup commit...thanks for the review @kidder !

**Comment by nilsdeppe** [comment]
This seems like a lot of code all to avoid holding the analytic data/solution in the DataBox when doing numerical initial data. Maybe the `Analytic` boundary conditions should hold the solution/data as a member variable instead? I wouldn't be surprised if a `DirichletKerrSchild` BC were added next, adding something like another 800 lines of code.

**Comment by geoffrey4444** [comment]
> This seems like a lot of code all to avoid holding the analytic data/solution in the DataBox when doing numerical initial data. Maybe the `Analytic` boundary conditions should hold the solution/data as a member variable instead? I wouldn't be surprised if a `DirichletKerrSchild` BC were added next, adding something like another 800 lines of code.

This is only there for debugging BBH BCs...once Bjorhus works, I plan to remove this. But getting it merged will make it more feasible for my students to help me try different scenarios

**Review by kidder — COMMENTED** [review]
you can squash

**Comment by nilsdeppe** [comment]
@geoffrey4444 can you rebase and remove `noexcept`s. We should be good to merge this Wednesday (Larry will be back then) assuming the `noexcept`s are removed :)

**Comment by geoffrey4444** [comment]
Is there anything else you'd like me to do with this PR? I thought I got the noexcepts...

**Review by kidder — APPROVED** [review]

**Pull request merged**